addons/account/models/account_tax.py

```diff
@@ -1635,9 +1627,10 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
             }
             tax_reps_data = tax_data['tax_reps_data'] = []
             for tax_rep in tax_reps:
+                tax_amount_currency = tax_data.get('tax_amount_currency', tax_data.get('raw_tax_amount_currency'))
```

Comment:
I don't like this but `_add_accounting_data_to_base_line_tax_details` is being called with base_lines in a rounded and non-rounded state, so without further refactoring it's the only way I see to keep current behaviour

For reference `AccountTax.compute_all` calls `_add_accounting_data_to_base_line_tax_details` before adding rounded tax details. And everywhere else `_add_accounting_data_in_base_lines_tax_details` is called with base lines with rounded data.